pkg/config/autodiscover/autodiscover.go

```diff
@@ -67,3 +67,63 @@ func makeGetCommand(resourceType, labelQuery string) *exec.Cmd {
 
 	return cmd
 }
+
+// getContainersByLabel builds `config.Container`s from containers in pods matching a label.
+func getContainersByLabel(label configsections.Label) (containers []configsections.Container, err error) {
+	pods, err := GetPodsByLabel(label)
+	if err != nil {
+		return nil, err
+	}
+	for i := range pods.Items {
+		containers = append(containers, buildContainersFromPodResource(&pods.Items[i])...)
+	}
+	return containers, nil
+}
+
+// getContainerIdentifiersByLabel builds `config.ContainerIdentifier`s from containers in pods matching a label.
+func getContainerIdentifiersByLabel(label configsections.Label) (containerIDs []configsections.ContainerIdentifier, err error) {
+	containers, err := getContainersByLabel(label)
+	if err != nil {
+		return nil, err
+	}
+	for _, c := range containers {
+		containerIDs = append(containerIDs, c.ContainerIdentifier)
+	}
+	return containerIDs, nil
+}
+
+// getContainerByLabel returns exactly one container with the given label. If any other number of containers is found
+// then an error is returned along with an empty `config.Container`.
+func getContainerByLabel(label configsections.Label) (container configsections.Container, err error) {
```

Comment:
question: if we scale a pod to more than one replicas, then the labels will be shared and used by all the replicas.
This function would fail at that point.
in which context we expect to have only one replicas all the time ?

Reply:
this is used to find the partner containers, i.e. orchestrator and fs-diff, not on targets. This actually could fail with only one pod too if it contains more than 1 containers.